gbpservice/nfp/modules/configurator.py

```diff
@@ -0,0 +1,458 @@
+        #                  *aaS response RPC method name>,
+        #        'kwargs': [{<data1>}, {data2}]
+        #    },
+        #    {
+        #    }, ...
+        # ]
+        #
+        # Initially, notification data will be empty and is populated
+        # after processing each request data in the request data list
+        notification_data = {}
+
+        # Handover the request data list and notification data to the
+        # identified service agent
+        sa_obj.forward_request(sa_info_list, notification_data)
+
+    def create_network_device_config(self, context, request_data):
+        """RPC method to configure a network service device.
+
+        Configures a network service VM to facilitate network service
+        operation. This RPC method is invoked by the configurator REST
+        server. It configures a network service based on the configuration
+        request specified in the request_data argument.
+
+        :param context: RPC context instance
+        :param request_data: RPC data
+
+        Returns: None
+
+        """
+
+        try:
+            self._send_request('create', request_data)
+        except Exception as err:
+            msg = ("Failed to create network device configuration. %s" %
+                   str(err).capitalize())
+            LOG.error(msg)
+
+    def delete_network_device_config(self, context, request_data):
+        """RPC method to clear configuration of a network service device.
+
+        Clears configuration of a network service VM. This RPC method is
+        invoked by the configurator REST server. It clears configuration
+        of a network service based on the configuration request specified
+        in the request_data argument.
+
+        :param context: RPC context instance
+        :param request_data: RPC data
+
+        Returns: None
+
+        """
+
+        try:
+            self._send_request('delete', request_data)
+        except Exception as err:
+            msg = ("Failed to delete network device configuration. %s" %
+                   str(err).capitalize())
+            LOG.error(msg)
+
+    def update_network_device_config(self, context, request_data):
+        """RPC method to update of configuration in a network service device.
+
+        Updates configuration of a network service VM. This RPC method is
+        invoked by the configurator REST server. It updates configuration
+        of a network service based on the configuration request specified
+        in the request_data argument.
+
+        :param context: RPC context instance
+        :param request_data: RPC data
+
+        Returns: None
+
+        """
+
+        try:
+            self._send_request('update', request_data)
+        except Exception as err:
+            msg = ("Failed to update network device configuration. %s" %
+                   str(err).capitalize())
+            LOG.error(msg)
+
+    def create_network_service_config(self, context, request_data):
```

Comment:
Sync up with Magesh, be consistent with the names in orchestrator. network_service is network_function.